contracts/feature-tests/composability/forwarder-raw/src/forwarder_raw.rs

```diff
@@ -132,50 +132,98 @@ pub trait ForwarderRaw {
             .transfer_execute();
     }
 
-    #[view]
-    #[storage_mapper("callback_data")]
-    fn callback_data(
+    #[endpoint]
+    fn forward_async_retrieve_multi_transfer_funds(
         &self,
-    ) -> VecMapper<(
-        TokenIdentifier,
-        BigUint,
-        ManagedVec<Self::Api, ManagedBuffer>,
-    )>;
+        to: ManagedAddress,
+        #[var_args] token_payments: ManagedVarArgs<MultiArg3<TokenIdentifier, u64, BigUint>>,
+    ) {
+        let mut arg_buffer = ManagedArgBuffer::new_empty(self.raw_vm_api());
+        for multi_arg in token_payments.into_iter() {
+            let (token_identifier, token_nonce, amount) = multi_arg.into_tuple();
 
-    #[view]
-    fn callback_data_at_index(
+            arg_buffer.push_arg(token_identifier);
+            arg_buffer.push_arg(token_nonce);
+            arg_buffer.push_arg(amount);
+        }
+
+        self.raw_vm_api().async_call_raw(
+            &to,
+            &self.types().big_uint_zero(),
+            &self
+                .types()
+                .managed_buffer_from(&b"retrieve_multi_funds_async"[..]),
+            &arg_buffer,
+        );
+    }
+
+    #[endpoint]
+    fn forwarder_async_send_and_retrieve_multi_transfer_funds(
         &self,
-        index: usize,
-    ) -> MultiResult3<TokenIdentifier, BigUint, ManagedVec<Self::Api, ManagedBuffer>> {
-        let (token, payment, args) = self.callback_data().get(index);
-        (token, payment, args.into()).into()
+        to: ManagedAddress,
+        #[var_args] token_payments: ManagedVarArgs<MultiArg3<TokenIdentifier, u64, BigUint>>,
+    ) {
+        let mut all_payments = Vec::new();
+        for multi_arg in token_payments.into_iter() {
+            let (token_identifier, token_nonce, amount) = multi_arg.into_tuple();
+
+            all_payments.push(EsdtTokenPayment {
+                token_identifier,
+                token_nonce,
+                amount,
+                token_type: EsdtTokenType::Invalid // ignored
+            });
+        }
+
+        self.send().transfer_multiple_esdt_via_async_call(
+            &to,
+            &all_payments.managed_into(self.raw_vm_api()),
+            &b"burn_and_create_retrive_async"[..],
+        );
     }
 
+    #[view]
+    #[storage_mapper("callback_data")]
+    fn callback_data(&self) -> VecMapper<ManagedVec<Self::Api, ManagedBuffer>>;
+
+    #[storage_mapper("callback_payments")]
+    fn callback_payments(&self) -> VecMapper<(TokenIdentifier, u64, BigUint)>;
```

Comment:
not critical, but more preferable to see a struct rather than a tuple of 3+ elements